app/assets/stylesheets/components/_slot-picker.scss

```diff
@@ -0,0 +1,362 @@
+    }
+  }
+}
+
+.slot-picker-times__action {
+  background: none;
+  cursor: pointer;
+  border: 1px solid $color-black;
+  width: 98%;
+  margin: 0 0 16px;
+  font-size: 1em;
+  padding: 8px;
+
+  @include media(tablet) {
+    width: 75%;
+  }
+
+  &:hover,
+  &:focus {
+    background: $calendar-selection;
+  }
+}
+
+.slot-picker-time-header {
+  margin-top: 0;
+}
+
+.slot-picker-nav {
+  display: flex;
+  box-shadow: -1px -1px $calendar-border,
+    inset -1px 0 $calendar-border;
+}
+
+.slot-picker-header {
+  margin-top: 0;
+}
+
+.slot-picker-nav__item {
+  flex-basis: 33.333%;
+  margin: 8px 0;
+  text-align: center;
+  font-size: 18px;
+  line-height: 32px;
+
+  @include media(tablet) {
+    font-size: 24px;
+  }
+
+  .js-enabled & {
+    margin: 10px 0;
+  }
+}
+
+.slot-picker-nav__item--prev {
+  text-align: left;
+}
+
+.slot-picker-nav__item--next {
+  text-align: right;
+}
+
+.slot-picker-nav-button {
+  height: 32px;
+  width: 100%;
+  font-size: .8em;
+  cursor: pointer;
+  border: 0;
+  background-color: transparent;
+  background-repeat: no-repeat;
+  background-size: 32px;
+
+  &:hover {
+    text-decoration: underline;
+  }
+}
+
+.slot-picker-nav-button--prev {
+  background-image: url(data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2263%22%20height%3D%2263%22%20viewBox%3D%220%200%2063%2063%22%3E%3Ccircle%20fill%3D%22%23DEE0E2%22%20cx%3D%2231.5%22%20cy%3D%2231.5%22%20r%3D%2231%22%2F%3E%3Cpath%20fill%3D%22%2370787C%22%20d%3D%22M17.527%2031.038l18.535-18.536%205.41%205.41-13.28%2013.28%2013.28%2013.278-5.41%205.41-18.535-18.536.154-.153%22%2F%3E%3C%2Fsvg%3E);
+  background-position: 4% center;
+  padding-left: 42px;
+  text-align: left;
+}
+
+.slot-picker-nav-button--next {
+  background-image: url(data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20width%3D%2263%22%20height%3D%2263%22%20viewBox%3D%220%200%2063%2063%22%3E%3Ccircle%20fill%3D%22%23DEE0E2%22%20cx%3D%2231.5%22%20cy%3D%2231.5%22%20r%3D%2231%22%2F%3E%3Cpath%20fill%3D%22%2370787C%22%20d%3D%22M45.473%2031.038L26.938%2012.502l-5.41%205.41%2013.28%2013.28-13.28%2013.278%205.41%205.41%2018.535-18.536-.154-.153%22%2F%3E%3C%2Fsvg%3E);
+  background-position: 96% center;
+  padding-right: 42px;
+  text-align: right;
+}
+
+.slot-picker-nav-button__short-text {
+  display: inline;
+
+  @include media(desktop) {
+    display: none;
+  }
+}
+
+.slot-picker-nav-button__long-text {
+  display: none;
+
+  @include media(desktop) {
+    display: inline;
+  }
+}
+
+.slot-picker-calendar__day-text {
+  position: absolute;
+  top: 5px;
+  left: 5px;
+}
+
+.slot-picker-key {
+  margin: 16px 0 32px;
+  list-style: none;
+}
+
+.slot-picker-key__item {
+  list-style: none;
+  display: inline-block;
+  margin: 0 8px 0 0;
+
+  &:before {
+    content: "";
+    display: inline-block;
+    width: 31px;
+    height: 31px;
+    margin-right: 8px;
+    border: 1px solid $grey-3;
```

Comment:
Added - wasn't sure about the white and black colours I've used throughout though?